src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -540,22 +529,12 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
                     adapter.InvalidateCache();
                 }
 
-                Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>();
-
-                foreach (KeyValuePair<string, ConfigurationSetting> kvp in data)
-                {
-                    ConfigurationSetting setting = kvp.Value;
-                    foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.UserDefinedMappers)
-                    {
-                        setting = func(setting).Result;
-                    }
-                    mappedData[kvp.Key] = setting;
-                }
+                Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationData(data);
 
                 try
                 {
                     SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
-                    _applicationSettings = dataCopy;
+                    _applicationSettings = data.ToDictionary(kvp => kvp.Key, kvp => new ConfigurationSetting(kvp.Value.Key, kvp.Value.Value, kvp.Value.Label, kvp.Value.ETag));

```

Comment:
If we call `MapConfigurationData` on `data` before deep copying it, `data` has already been modified by user defined mappers and we lose the original ConfigurationSetting returned by server.